iceoryx_examples/icedelivery/iox_publisher_with_options.cpp

```diff
@@ -29,6 +29,7 @@ static void sigHandler(int f_sig [[gnu::unused]])
 {
     // caught SIGINT or SIGTERM, now exit gracefully
     killswitch = true;
+    iox::runtime::PoshRuntime::getInstance().shutdown();
```

Comment:
Maybe have a comment here that describes that the shutdown is optional but makes sense here to ensure that the blocking publisher is released